engines/python/setup/djl_python/sagemaker.py

```diff
@@ -0,0 +1,203 @@
+import yaml
+import io
+import logging
+from pathlib import Path
+from djl_python import Input
+from djl_python import Output
+
+
+class SageMakerInferenceService(object):
+    def __init__(self) -> None:
+        self.inference_spec = None
+        self.model_dir = None
+        self.model = None
+        self.schema_builder = None
+        self.inferenceSpec = None
+        self.metadata = None
+        self.default_serializer = None
+        self.default_deserializer = None
+        self.initialized = False
+
+    def load_yaml(self, path: str):
+        with open(path, mode="r") as file:
+            return yaml.full_load(file)
+
+    def load_metadata(self):
+        metadata_path = Path(self.model_dir).joinpath('metadata.yaml')
+        return self.load_yaml(metadata_path)
+    
+    def load_and_validate_pkl(self, path, hash_tag):
+        import os
+        import hmac
+        import hashlib
+        import cloudpickle
+        with open(path, mode="rb") as file:
+            buffer = file.read()
+            secret_key = os.getenv('SAGEMAKER_SERVE_SECRET_KEY')
+            stored_hash_tag = hmac.new(secret_key.encode(), msg=buffer, digestmod=hashlib.sha256).hexdigest()
+            if not hmac.compare_digest(stored_hash_tag, hash_tag):
+                raise Exception("Object is not valid: " + path)
+            
+        with open(path, mode="rb") as file:
+            return cloudpickle.load(file)
+        
+    def load_default_schema(self, path):
+        schema = self.load_yaml(path=path)
+        if "input_deserializer" in schema:
+            deserializer_name = schema.get("input_deserializer")
+            if deserializer_name == "BytesDeserializer":
+                from sagemaker.deserializers import BytesDeserializer
```

Comment:
This logic may be cleaner inside PySDK as a `get_input_deserializer(deserializer_name)` function. Also helps for sharing code with other model servers. I notice this with much of the other parts of this file. This might be a refactor to consider moving forwards and as work with other model servers increases, but I think it is fine to merge